extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/sharing/HashSharedConnectionMap.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.auth.jdbc.sharing;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * A HashMap-based implementation of the SharedConnectionMap.
+ *
+ * @author Michael Jumper
+ */
+public class HashSharedConnectionMap implements SharedConnectionMap {
+
+    /**
+     * Keeps track of the share key to SharedConnectionDefinition mapping.
+     */
+    private final ConcurrentMap<String, SharedConnectionDefinition> connectionMap =
+            new ConcurrentHashMap<String, SharedConnectionDefinition>();
+
+    @Override
+    public SharedConnectionDefinition get(String key) {
+        
+        // There are no null share keys
+        if (key == null)
+            return null;
+
+        // Update the last access time and return the SharedConnectionDefinition
+        return connectionMap.get(key);
+
+    }
+
+    @Override
+    public void put(String key, SharedConnectionDefinition definition) {
+        connectionMap.put(key, definition);
```

Comment:
Well, the other two methods do a `null` check so they can safely return that the key cannot possibly be present, as otherwise even attempting to do a lookup with a `null` key on a `ConcurrentHashMap` will result in a `NullPointerException`.

What would be the safe behavior here for such a `null` check? Silently don't store anything? Sounds like an NPE would be preferable (which is what will currently happen thanks to `ConcurrentHashMap`).